### ISSUE TITLE
Latest / Fresh plugins list is not updated (?)

### ISSUE THREAD

**Issue by agiudiceandrea** [qgis/QGIS-Django#82]
Currently, the following are the first plugins in the Latest / Fresh plugin list

  | Name |   |   | Author | Created on | Stars (votes) | Stable | Exp.
-- | -- | -- | -- | -- | -- | -- | -- | --
  | Land Productivity Analysis Tool (LPAT) | — | 409 | RCMRD | July 14, 2020 | (1) | 0.11 | —
  | AGT - Archaeological Geophysics Toolbox | — | 19183 | INRAP - Guillaume Hulin, Francois-Xavier Simon, Nariman Hatami | Aug. 29, 2017 | (30) | 3.1.0 | —
  | Lizmap |   | 95525 | 3Liz | Feb. 16, 2012 | (95) | 3.2.17 | 3.2.0
  | CadTools | — | 216131 | Stefan ZIegler | Feb. 14, 2012 | (264) | 0.6.3 | —
  | Spline | — | 32414 | Radim Blazek | Feb. 5, 2014 | (37) | 0.1.6 | 0.1.3

However, they are not the latest or "fresher" plugins as you can see at https://plugins.qgis.org/plugins/?sort=-created_on

  | Name |   |   | Author | Created on ↓ | Stars (votes) | Stable | Exp.
-- | -- | -- | -- | -- | -- | -- | -- | --
  | Coordinate Capture | — | 23 | Stefanos Natsis | today | (0) | 0.2 | —
  | Digitize local coordinates | — | 47 | Morten Sickel | Aug. 6, 2020 | (0) | — | 0.1
  | Manejo | — | 23 | Carvalho Neto, R.M. and Sustembio Environmental Services | Aug. 5, 2020 | (1) | — | 0.1
  | JSON to Waterbodies for QGIS | — | 105 | Alexandra Crowe | Aug. 4, 2020 | (0) | 0.1 | —
  | Pobieracz danych GUGiK | — | 564 | EnviroSolutions Sp. z o.o. - Michał Włoga | Aug. 3, 2020 | (15) | 0.2.1 | —
  | Projekt-Check | — | 51 | GGR-Planung, HafenCity University Hamburg, ILS Dortmund | July 31, 2020 | (0) | 1.1 | —
  | SLYR (Community Edition) | — | 2006 | Nyall Dawson | July 31, 2020 | (1) | 3.0.6 | —
  | Lines Ranking | — | 68 | Julia Borisova, Mikhail Sarafanov | July 30, 2020 | (2) | 1.0 | —
  | Altibase QGIS Plugin | — | 47 | Altibase Corp. | July 29, 2020 | (0) | — | 0.1
  | ANA Data Acquisition | — | 104 | HGE-IPH | July 26, 2020 | (0) | — | 0.1

**Comment by timlinux** [comment]
Latest plugins are shown on the landing page:

![image](https://user-images.githubusercontent.com/178003/94548321-70e6b200-0248-11eb-8ed8-0abf6ce5bafc.png)


https://plugins.qgis.org/

The list should be the 10 newest plugin updates (either newly uploaded for the first time or recently updated)

The fresh plugins at https://plugins.qgis.org/plugins/fresh/ should list the plugins in reverse chronological order based on the newest version / update that has been uploaded.

**Comment by rduivenvoorde** [comment]
Can this be closed? Or is this still an issue?

**Comment by agiudiceandrea** [comment]
It seems still an issue.

Current list of Latest plugins = Fresh plugins:

Name | Latest version
------ | ----------------
Layer Tree Icons | Oct. 26, 2020
BLN Import | Aug. 31, 2017
Cartography Tools | Oct. 25, 2020
Split Features On Steroids | March 29, 2018
Floodrisk | Aug. 21, 2020
Semi-Automatic Classification Plugin | Oct. 25, 2020
QuerySelection | Nov. 30, 2019
QCopycanvas | Oct. 25, 2020
qtranus | March 21, 2020
Proportional circles | Feb. 14, 2018

**Comment by timlinux** [comment]
@sumandari please investigate and fix.

**Comment by sumandari** [comment]
Latest plugins list in template
https://github.com/qgis/QGIS-Django/blob/b887e3168b259d2c1a5bfcc127e0290bf902036a/qgis-app/templates/flatpages/homepage.html#L20
the `fresh` object list is:

https://github.com/qgis/QGIS-Django/blob/62ee4e2fc0450399c211fcc4d79fc8235702be34/qgis-app/homepage.py#L13

It has same query manager with `Fresh` page list:
https://github.com/qgis/QGIS-Django/blob/62ee4e2fc0450399c211fcc4d79fc8235702be34/qgis-app/plugins/urls.py#L40

so that the Latest plugins will displays fresh plugins limited by first 10 query results.

`fresh_objects` query manager is ordered by `-modified-on`, it will order plugins with **last modified date**. Where when a new plugin uploaded, the `last modified` date will same with `created on`
https://github.com/qgis/QGIS-Django/blob/62ee4e2fc0450399c211fcc4d79fc8235702be34/qgis-app/plugins/models.py#L92
https://github.com/qgis/QGIS-Django/blob/62ee4e2fc0450399c211fcc4d79fc8235702be34/qgis-app/plugins/models.py#L350

---
TL;DR
The plugins latest plugins will display first 10 fresh plugins. Where it's ordered by `-modified-on`.

**Comment by sumandari** [comment]
The latest deployment on 10th November will display the last modified date beside create date




Name| Last modified | Created on
-- | -- | --
Wtyczka GIS Support | today	 | Jan. 23, 2020	
Sentinel-2 Download | today | March 16, 2020	
Data-Driven Input Mask | today | May 8, 2013	
PAT - Precision Agriculture Tools| yesterday | Feb. 5, 2019	
EDE Interpolation | yesterday | April 21, 2019	
QPackage | Yesterday | Aug. 20, 2015	


![plugins_ticket82_fig1](https://user-images.githubusercontent.com/40058076/100205145-aefa0c80-2f3f-11eb-9d3d-0ec1d8c1dd13.png)

@agiudiceandrea please kindly advise if this table display meet your expectation or you have another preference ?

**Comment by agiudiceandrea** [comment]
Thank you, @sumandari. Anyway the issue remains...

There is a "Latest plugins" list and a "Fresh plugins" list: currently both the lists contains always the same items in the same order.
So, I think the lists should have the same name, or they should list the plugins in different ways...

But, more important, actually the current "Latest plugins" / "Fresh plugins" list is still weird. I think a user wants to know, the newer plugins added for the first time and the plugins recently updated, i.e the one that have a new version available.

Now the list is:

Name | Latest plugin version | Created on
-- | -- | --
Wtyczka GIS Support | Nov. 24, 2020 | Jan. 23, 2020
Sentinel-2 Download | April 13, 2020 | March 16, 2020
Data-Driven Input Mask | Nov. 25, 2020 | May 8, 2013
PAT - Precision Agriculture Tools | Nov. 24, 2020 | Feb. 5, 2019
EDE Interpolation | June 5, 2020 | April 21, 2019
QPackage | Feb. 5, 2018 | Aug. 20, 2015
KML Tools | Sept. 7, 2020 | Sept. 7, 2018
Shape Tools | July 2, 2020 | March 10, 2017
QuickMapServices | Dec. 2, 2019 | Jan. 11, 2015
OpenLayers Plugin | April 9, 2018 | Oct. 4, 2012

I don't see any order in this list. Maybe the "last modified" date isn't suitable for a meaningful sorting...

**Comment by sumandari** [comment]
Aha! Latest plugin version. Thanks @agiudiceandrea. I submitted PR #115 for displaying the latest plugin version. But it doesn't change the list of both Latest and Fresh plugins yet. Please have a look at it.
@timlinux @dimasciput could you give suggestion about how the Latest Plugins and Fresh Plugins list should be.

**Comment by agiudiceandrea** [comment]
I think we could have a list (for plugins with newer versions) ordered descending by the date of the latest plugin version published and a list (for newly uploaded plugins) ordered descending by the created on date.

**Comment by sumandari** [comment]
@agiudiceandrea the new list for new plugins and updated plugins has been implemented by https://github.com/qgis/QGIS-Django/pull/115

**Comment by agiudiceandrea** [comment]
Fixed by #115
